OpenSim/Simulation/SimbodyEngine/Constraint.cpp

```diff
@@ -71,7 +71,29 @@ void Constraint::setNull(void)
  */
 void Constraint::constructProperties(void)
 {
-    constructProperty_isDisabled(false);
+    constructProperty_isEnforced(true);
+}
+
+void Constraint::updateFromXMLNode(SimTK::Xml::Element& node,
+                                   int versionNumber) {
+    if(versionNumber < XMLDocument::getLatestVersion()) {
+        if(versionNumber < 30508) {
+            // Rename property 'isDisabled' to 'isEnforced' and
+            // negate the contained value.
+            std::string oldName{"isDisabled"};
+            std::string newName{"isEnforced"};
+            if(node.hasElement(oldName)) {
+                auto elem = node.getRequiredElement(oldName);
+                elem.setElementTag(newName);
+                if(elem.getValue().find("true") != std::string::npos)
+                    elem.setValue("false");
+                else if(elem.getValue().find("false") != std::string::npos)
+                    elem.setValue("true");
```

Comment:
Agreed. I'm fine with leaving it for now and revisiting once we have schema-driven error checking for model files. I think the current functionality is fine:

| if the old value is… | then the new value will be… |
| --- | --- |
| true | false |
| false | true |
| fales | |
| giraffe | |

Note that values not in {"true", "false"} will not be written to the new model file, so a typo like `<isDisabled>fales</isDisabled>` will **not** be converted to `<isEnforced>fales</isEnforced>`, which would be bad.